releasenotes/notes/improve-hls-plugins-for-clifford-t-96677c199ee88989.yaml

```
@@ -0,0 +1,26 @@
+---
+features_transpiler:
+  - |
+    Added a new class :class:`.OptimizationMetric` which specifies the optimization
+    criterion during transpilation. Currently the two supported metrics are:
+
+    * ``COUNT_2Q``: optimizes the number of two-qubit gates,
+    * ``COUNT_T``: optimizes the number of T-gates, when transpiling into a Clifford+T
+      basis set.
+
+    This class is used by the :class:`.HighLevelSynthesis` transpiler pass
+    to select decompositions optimized either for the two-qubit-gate count or the
+    T-count.
```

How this is currently formulated I would expect that I can pass the optimization metric into `transpile`, but it's more specific right now. Could we reformulate this to reflect that? Maybe something like the following:
```suggestion
    Added a new class :class:`.OptimizationMetric` which specifies the optimization
    criterion in the :class:`.HighLevelSynthesis` pass. Currently the two 
    supported metrics are:

    * ``COUNT_2Q``: optimizes the number of two-qubit gates,
    * ``COUNT_T``: optimizes the number of T-gates, when transpiling into a Clifford+T
      basis set.

    The transpiler automatically selects the target metric based on the basis gate set, e.g.
    it will use ``COUNT_T`` if a Clifford+T basis set is detected. This argument cannot yet
    be set manually at the level of :func:`~qiskit.compiler.transpile`.
```

---

Thanks, this is indeed a better way to present the changes, did in [1a91fee](https://github.com/Qiskit/qiskit/pull/14902/commits/1a91fee5efae8d8b02368215ab72aa2f1e105827).